ArduCopter/mode_guided.cpp

```diff
@@ -768,6 +780,159 @@ bool ModeGuided::limit_check()
     return false;
 }
 
+//loiter turns(circle) mode
+
+void ModeGuided::do_circle(const mavlink_command_long_t &packet)
+{
+    const Location circle_center = loc_from_param(packet);
+
+    // calculate radius
+    uint8_t circle_radius_m = radius_from_param(packet); // circle radius recived
+
+    //store the number of turns
+    turn_count = (uint8_t)packet.param1;
+
+    // move to edge of circle (verify_circle) will ensure we begin circling once we reach the edge
+    circle_movetoedge_start(circle_center, circle_radius_m);
+}
+
+Location ModeGuided::loc_from_param(const mavlink_command_long_t &packet)
+{    
+    Location ret((int32_t)packet.param5,(int32_t)packet.param6,(int32_t)packet.param7,copter.current_loc.get_alt_frame());
```

Comment:
This doesn't look right for a command-long - those values are usually (e.g) degrees-lat and locations are deglat*1e7